### PR TITLE
Teach FML that Swift's optional chaining is different to Kotlin's optional chaining

### PR DESCRIPTION
**Pull request by jhugman**
During initial phases of messaging work, @nishant2718 found that `Map<String, String>` produced invalid Swift code. Works fine for `Map<Enum, String>` and `Map<String, T>`.

This PR adds tests and a fix. Issue was Swift's optional chaining is different to Kotlin's safe call.

A A-S release needs to happen before iOS can use this. This does not affect application services code _shipped_ with the app, just tooling.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Review by travis79 — APPROVED** [review]

**Review by tarikeshaq — APPROVED** [review]
Thanks @jhugman!!! I can cut the release to get this to iOS

**Comment by tarikeshaq** [comment]
Not sure why github isn't catching that the coverage job is done. but it is done in circleci, we can merge this safely, all checks passed.

File: components/support/nimbus-fml/src/backends/swift/gen_structs/mod.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ impl ConcreteCodeOracle {
             TypeIdentifier::EnumMap(ref k_type, ref v_type) => {
                 Box::new(structural::MapCodeType::new(k_type, v_type))
             }
-            _ => unimplemented!(),
+            _ => unimplemented!("Text and Image are to be implemented in EXP-2147"),
```
Comment (nishant2718) [inline components/support/nimbus-fml/src/backends/swift/gen_structs/mod.rs:127]: 🙏

File: components/support/nimbus-fml/src/backends/swift/gen_structs/common.rs [hunk 2]
```diff
@@ -38,17 +38,19 @@ pub(crate) mod code_type {
...
+        // We need to be quite careful about option chaining.
+        // Swift takes the `?` as an indicator to _stop evaulating the chain expression_ if the immediately preceeding
+        // expression returns an optional.
+        // Only the value_getter returns an optional, so that's all we need to `?`.
+        // https://docs.swift.org/swift-book/LanguageGuide/OptionalChaining.html
+        let getter = match (mapper, merger) {
+            (Some(mapper), Some(merger)) => format!("{}?.{}.{}", getter, mapper, merger),
+            (Some(mapper), None) => format!("{}?.{}", getter, mapper),
+            (None, Some(merger)) => format!("{}?.{}", getter, merger),
+            (None, None) => getter,
```
Comment (nishant2718) [inline components/support/nimbus-fml/src/backends/swift/gen_structs/common.rs:53]: 🙌

**Pull request merged**